modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/VolatileTxStateMetaStorage.java

```diff
@@ -171,39 +170,34 @@ public CompletableFuture<Void> vacuum(
                 } else if (TxState.isFinalState(meta0.txState())) {
                     Long initialVacuumObservationTimestamp = meta0.initialVacuumObservationTimestamp();
```

Comment:
why still `Long`? maybe make `initialVacuumObservationTimestamp()` not nullable and add `requireNonNull` here?

Reply:
It still can be null for not final tx states.